### PR TITLE
D165 - summary site numbers aggregation for duplicate sites with close (lat/lon) within 150m

### PR DESCRIPTION
**Pull request by tienhoah**

### CODE REVIEW

File: src/components/MermaidDash.js [hunk 1]
```diff
@@ -392,6 +394,30 @@ class MermaidDash extends Component {
...
+
+  getUniqueSiteCount(array) {
+    let duplicateCount = 0;
+    const totalSitesCount = array.length;
+    const coordinatesArr = array.map(({ geometry: { coordinates } }) => {
+      return coordinates;
+    });
+
+    for (let i = 0; i < totalSitesCount; i++) {
+      if (i !== totalSitesCount - 1) {
```
Comment (brianbancroft) [inline src/components/MermaidDash.js]: Can this be coalesced into `for (let i = 0; i < totalSitesCount - 1; i++) {`?

File: src/components/MermaidDash.js [hunk 2]
```diff
@@ -392,6 +394,30 @@ class MermaidDash extends Component {
...
+        for (let y = i + 1; y < totalSitesCount; y++) {
```
Comment (brianbancroft) [inline src/components/MermaidDash.js]: For the sake of convention and not to confuse y with an (x,y)-pairing, you might want to use `j`, and then `k` if you need something else.

File: src/components/MermaidDash.js [hunk 3]
```diff
@@ -392,6 +394,30 @@ class MermaidDash extends Component {
...
+          if (this.calDistance(coordinatesArr[i], coordinatesArr[y]) < 150) {
```
Comment (brianbancroft) [inline src/components/MermaidDash.js]: I suggest adding `// Assumes coordinates that are within 150m of each other are duplicate` for clarity just before this line

File: src/components/MermaidDash.js [hunk 4]
```diff
@@ -392,6 +394,30 @@ class MermaidDash extends Component {
...
+        }
+      }
+    }
+
+    return totalSitesCount - duplicateCount;
+  }
+
   getTransectCount(array, key) {
     const protocols = array.map(item => {
       return item.properties[key];
```
Comment (brianbancroft) [inline src/components/MermaidDash.js]: I know this is not part of this PR, but did you know that is the same as:

```js
const protocols = array.map(item => item.properties[key])
```

🌮  - Don't have to make a change here. Just a fun fact!

**Comment by tienhoah** [comment]
closes #165

**Pull request merged**